File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/home/HomeStatsContract.java
package ru.strcss.projects.moneycalc.moneycalcandroid.home;

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter;
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView;

/**
 * This specifies the contract between the view and the HomeStatsPresenter.
 */
public interface HomeStatsContract {
    interface View extends BaseView<Presenter> {

        void showErrorMessage(String msg);
//        void setTodayBalance(double todayBalance);
//        void setSummaryBalance(double summaryBalance);
//        void setSpendAll(int spendAll);
//        void setLeftAll(int leftAll);
    }

    interface Presenter extends BasePresenter<View> {

    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/home/HomeStatsContract.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.home

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView

/**
 * This specifies the contract between the view and the HomeStatsPresenter.
 */
interface HomeStatsContract {
    interface View : BaseView<Presenter> {

        override fun showErrorMessage(msg: String)
        //        void setTodayBalance(double todayBalance);
        //        void setSummaryBalance(double summaryBalance);
        //        void setSpendAll(int spendAll);
        //        void setLeftAll(int leftAll);
    }

    interface Presenter : BasePresenter<View>
}
